include/mbgl/map/map.hpp

```diff
@@ -34,6 +34,7 @@ class Map : private util::noncopyable {
     explicit Map(View&, FileSource&,
                  MapMode mapMode = MapMode::Continuous,
                  GLContextMode contextMode = GLContextMode::Unique,
+                 ViewportMode viewportMode = ViewportMode::Default,
```

Comment:
Rationale is that properties that affect code rendering gets higher priority in order of appearance - but I don't hold a strong opinion on this.